astropy/io/registry.py

```diff
@@ -77,10 +77,11 @@ def get_formats(data_class=None):
         rows.append((format_class[1].__name__, format_class[0], has_read,
                      has_write, has_identify, deprecated))
 
-    data = list(zip(*rows)) if rows else None
+    data = None
+    if rows:
```

Comment:
Maybe a silly stylistic comment, but I would do
```
if rows:
    data = ...
else:
    data = None
```
I think it shows the flow of the program more clearly.